src/components/Modals/WorldPermissionsModal/WorldPermissionsModal.tsx

```diff
@@ -0,0 +1,255 @@
+          !newCollaboratorUserList.has(wallet) && newCollaboratorUserList.add(wallet.toLowerCase())
+        })
+      }
+      return [...newCollaboratorUserList]
+    })
+  }, [worldPermissions])
+
+  const profilesByWallets = useMemo(() => profiles, [profiles])
+
+  const handleAddUserToColaboratorList = useCallback(
+    (_event: React.MouseEvent<HTMLButtonElement, MouseEvent>, _data: ButtonProps) => {
+      if (newAddress === '') {
+        setShowAddUserForm(false)
+      }
+      setErrorInvalidAddress(false)
+      const isValidAddress = isValid(newAddress)
+      if (!isValidAddress) {
+        setErrorInvalidAddress(true)
+      }
+      if (isValidAddress && metadata.worldName && newAddress && !collaboratorUserList.includes(newAddress)) {
+        setCollaboratorUserList(prev => [...prev, newAddress.toLowerCase()])
+      }
+      onGetProfilesRequest([newAddress])
+    },
+    [metadata.worldName, worldPermissions, newAddress, collaboratorUserList]
+  )
+
+  const handleUserPermissionList = useCallback(
+    (_event: React.MouseEvent, data: CheckboxProps & { worldPermissionName: WorldPermissionNames; wallet: string }) => {
+      const isValidAddress = isValid(data.wallet)
+      if (!isValidAddress) {
+        setErrorInvalidAddress(true)
+      }
+
+      if (
+        isValidAddress &&
+        worldPermissions?.[data.worldPermissionName].type === WorldPermissionType.AllowList &&
+        !(worldPermissions[data.worldPermissionName] as AllowListPermissionSetting).wallets.includes(data.wallet.toLowerCase())
+      ) {
+        setErrorInvalidAddress(false)
+        onPutWorldPermissionsRequest(metadata.worldName, data.worldPermissionName, WorldPermissionType.AllowList, data.wallet.toLowerCase())
+      }
+      if (
+        worldPermissions?.[data.worldPermissionName].type === WorldPermissionType.AllowList &&
+        (worldPermissions[data.worldPermissionName] as AllowListPermissionSetting).wallets.includes(data.wallet.toLowerCase())
+      ) {
+        setErrorInvalidAddress(false)
+        onDeleteWorldPermissionsRequest(
+          metadata.worldName,
+          data.worldPermissionName,
+          WorldPermissionType.AllowList,
+          data.wallet.toLowerCase()
+        )
+      }
+    },
+    [metadata.worldName, worldPermissions, newAddress]
+  )
+
+  const handleChangeAccessPermission = useCallback(
+    (_event: React.FormEvent<HTMLInputElement>, data: CheckboxProps) => {
+      const nextPermissionType = !data.checked ? WorldPermissionType.AllowList : WorldPermissionType.Unrestricted
+      if (metadata.worldName && worldPermissions?.access.type !== nextPermissionType) {
+        onPostWorldPermissionsRequest(metadata.worldName, WorldPermissionNames.Access, nextPermissionType)
+      }
+    },
+    [metadata.worldName, worldPermissions]
+  )
+
+  const handleRemoveCollaborators = useCallback(
+    (_event: React.MouseEvent<HTMLButtonElement, MouseEvent>, data: ButtonProps & { wallet: string }) => {
```

Comment:
What do you think about returning if `metadata.worldName` doesn't exist? We would be covering the first two conditions (instead of repeating the check) and the `setCollaborUserList` that shouldn't be triggered if it doesn't exist:
```suggestion
    (_event: React.MouseEvent<HTMLButtonElement, MouseEvent>, data: ButtonProps & { wallet: string }) => {
    if(!metadata.worldName) {
      return
    }
```